speechbrain/lobes/models/huggingface_transformers/discrete_ssl.py

```diff
@@ -284,3 +330,49 @@ def forward(
             org_tokens, SSL_layers, deduplicates, bpe_tokenizers
         )
         return org_tokens, org_embedding, processed_tokens
+
+    def decode(self, tokens, SSL_layers=None):
+        """Takes an input waveform and return its corresponding waveform.
+
+        Arguments
+        ---------
+        tokens : torch.Tensor
+            A (Batch, codes, layers) tensor of discrete units
+        SSL_layers: List[int]:
+            determine which layers of SSL should be used by the vocoder.
+
+        Returns
+        -------
+        waveforms: torch.tensor
+            Batch of mel-waveforms [batch, time]
+        """
+
+        all_layer_ids = self.ssl_layer_ids
+        num_clusters = self.num_clusters[0]
+        offsets = torch.arange(
+            0,
+            len(all_layer_ids) * num_clusters,
+            num_clusters,
+            device=self.device,
+        )
+
+        layers = all_layer_ids
+        if SSL_layers is not None:
+            layers = SSL_layers
+
+        offset_idxes = [all_layer_ids.index(x) for x in layers]
+        offsets = offsets[offset_idxes]
+        tokens += offsets + 1
```

Comment:
This is not safe as it changes the input in-place, better `tokens = tokens + offsets + 1`